src/Lean/Widget/Basic.lean

```diff
@@ -18,6 +18,7 @@ functionality is purpose-specific to showing the contents of infoview popups.
 structure InfoWithCtx where
   ctx  : Elab.ContextInfo
   info : Elab.Info
+  children : PersistentArray InfoTree
   deriving Inhabited, TypeName
```

Comment:
Let's move this to `InfoUtils` and use it in `hoverableInfoAt?` and `locationLinksOfInfo` etc. The `deriving TypeName` should probably be left here though.

Reply:
Done, let me know what you think of the refactor.

Reply:
Hi, sorry to only comment on this now but `InfoWithCtx` is only really intended for use as a way of bundling data together in the RPC protocol so that interactive expressions and go-to-def in the infoview work. As Gabriel put it, it's a kind of "type popup + go-to-def thunk". Unless you need to do something with RPC here, it would be better not to move it and to maybe add another type with whatever semantics you need.

Reply:
Duplicating it wouldn't be hard, but since the type is opaque in the RPC protocol I'm not really seeing the motivation. It's a sensible bundle of data to have in the server in general, and the name fits as well. Even if we don't necessarily need `children` in the widgets.

Reply:
The type seems like a better fit for `Elab.InfoTree`, if anything because it was already hacky as far as the RPC interface is concerned — the outputs of delaboration are `Elab.Info`s but they should really be a much more restricted `Delab.Info`. But okay, since `InfoWithCtx` is opaque we can always keep it around for uses such as in this PR but replace it with something else in the RPC interface. Btw @rish987 you probably want to also make the output of `hoverableInfoAt?` and `termGoalAt?` be `InfoWithCtx`, document what the `children` field is, and change the docstring to note that it's used for more than just RPC.

Reply:
Done, thanks for the suggestions.